src/client/push-client.js

```diff
@@ -16,24 +16,24 @@ import SubscriptionFailedError from './subscription-failed-error';
 import PushClientEvent from './push-client-event';
 import EventDispatch from './event-dispatch';
 
-// document.currentScript is not supported in all browsers, but it IS supported
-// in all browsers that support Push.
-// TODO(mscales): Ensure that this script will not cause errors in unsupported
-// browsers.
-let currentScript = document.currentScript.src;
+// Set the default scope to be relative to the service worker
+// script with an addition string to prevent any unlikely collisions
+// This should allow multiple instances of push on the same scope to work
+const SCOPE = './goog.push.scope/';
 
-// Make the dummy service worker scope be relative to the library script. This
-// means that you can have multiple projects hosted on the same origin without
-// them interfering with each other, as long as they each use a different URL
-// for the script.
-const SCOPE = new URL('./goog.push.scope/', currentScript).href;
-const WORKER_URL = new URL('./worker.js', currentScript).href;
 const SUPPORTED = 'serviceWorker' in navigator &&
     'PushManager' in window &&
     'Notification' in window &&
     'permissions' in navigator &&
     'showNotification' in ServiceWorkerRegistration.prototype;
 
+const ERROR_MESSAGES = {
+  'bad constructor': 'The PushClient constructor expects either service ' +
+    'worker registration or the path to a service worker file and an ' +
+    'optional scope string.',
+  'redundant worker': 'Worker became redundant'
+};
+
 const registrationReady = function(registration) {
   if (registration.active) {
```

Comment:
This has been discussed offline, but for the record, the situation I was thinking about was where the old SW doesn't handle `push` events, but the new one does. This could lead to messages being dropped until the new SW is controlling the page. (For example, it's plausible the server will attempt to send a message like "Thanks for subscribing" or "Here's what a notification looks like" as soon as it receives subscription information from the client--the messages might not get through.)

(I think the conclusion was that there's no elegant way to deal with this--it just needs to be documented that some combinations of SW/client states may be problematic.)